src/main/java/org/dependencytrack/tasks/vulnerabilitypolicy/S3Client.java

```diff
@@ -0,0 +1,57 @@
+package org.dependencytrack.tasks.vulnerabilitypolicy;
+
+import alpine.Config;
+import alpine.common.logging.Logger;
+import io.minio.MinioClient;
+import org.dependencytrack.common.ConfigKey;
+
+public class S3Client {
+    private final String bundleToFetch;
+    private final String s3AccessKey;
+    private final String s3SecretKey;
+    private final String s3BucketName;
+    private MinioClient minioClient;
+
+    private String bundleFetchEndpoint;
+
+    private final String region;
+
+    public String getBundleToFetch() {
+        return bundleToFetch;
+    }
+
+    public String getS3BucketName() {
+        return s3BucketName;
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(S3Client.class);
+
+    public S3Client() {
+        this.bundleFetchEndpoint = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_BUNDLE_URL);
+        this.bundleToFetch = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_BUNDLE_NAME);
+        this.s3AccessKey = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_ACCESS_KEY);
+        this.s3SecretKey = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_SECRET_KEY);
+        this.s3BucketName = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_BUCKET_NAME);
+        this.region = Config.getInstance().getProperty(ConfigKey.VULNERABILITY_POLICY_S3_REGION);
+        if (bundleToFetch == null || s3BucketName == null || s3SecretKey == null || s3AccessKey == null) {
+            LOGGER.error("The required parameters for S3 file access have not been provided. Please check");
+        }
+    }
+
+    private void createClient() {
+        MinioClient.Builder builder = MinioClient.builder().endpoint(bundleFetchEndpoint)
+                .credentials(s3AccessKey, s3SecretKey);
+        if (this.region != null && !region.trim().equals("")) {
+            builder.region(region);
+        }
+        this.minioClient = builder.build();
+    }
+
+    public MinioClient getMinioClient() {
+        if (this.minioClient == null) {
+            createClient();
+        }
+        return minioClient;
+    }
```

Comment:
This is a bit of a leaky abstraction here. Users of the `S3Client` class should ideally never know that it uses MinIO behind the scenes. 

Would it be possible to implement the high-level functionality directly in `S3Client`, instead of handing out instances of `MinioClient`? For example, in `S3StorageHandler`, you do three things:

1. Check if a bucket exists
2. Check if a given object exists within that bucket, and return its ETag if it does
3. Download an object, and return it as byte array

So, theoretically, in `S3Client` you could expose these methods:

```java
public boolean doesBucketExist(String bucket) throws IOException {
    // TODO
}

public String getObjectEtag(String bucket, String objectName) throws IOException {
    // TODO
}

public byte[] getObject(String bucket, String objectName) throws IOException {
    // TODO
}
```